File: src/main/java/org/cubexmc/metro/manager/SelectionManager.java
package org.cubexmc.metro.manager;

import org.bukkit.Location;
import org.bukkit.entity.Player;

import java.util.Map;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public class SelectionManager {
    private final Map<UUID, Location> corner1Selections = new ConcurrentHashMap<>();
    private final Map<UUID, Location> corner2Selections = new ConcurrentHashMap<>();

    public void setCorner1(Player player, Location location) {
        corner1Selections.put(player.getUniqueId(), location);
    }

    public void setCorner2(Player player, Location location) {
        corner2Selections.put(player.getUniqueId(), location);
    }

    public Location getCorner1(Player player) {
        return corner1Selections.get(player.getUniqueId());
    }

    public Location getCorner2(Player player) {
        return corner2Selections.get(player.getUniqueId());
    }

    public boolean isSelectionComplete(Player player) {
        return corner1Selections.containsKey(player.getUniqueId()) && corner2Selections.containsKey(player.getUniqueId());
    }
}

File: src/main/java/org/cubexmc/metro/manager/SelectionManager.kt
package org.cubexmc.metro.manager

import java.util.UUID
import java.util.concurrent.ConcurrentHashMap
import org.bukkit.Location
import org.bukkit.entity.Player

class SelectionManager {
    private val corner1Selections: MutableMap<UUID, Location> = ConcurrentHashMap()
    private val corner2Selections: MutableMap<UUID, Location> = ConcurrentHashMap()

    fun setCorner1(player: Player, location: Location) {
        corner1Selections[player.uniqueId] = location
    }

    fun setCorner2(player: Player, location: Location) {
        corner2Selections[player.uniqueId] = location
    }

    fun getCorner1(player: Player): Location? = corner1Selections[player.uniqueId]

    fun getCorner2(player: Player): Location? = corner2Selections[player.uniqueId]

    fun isSelectionComplete(player: Player): Boolean =
        corner1Selections.containsKey(player.uniqueId) && corner2Selections.containsKey(player.uniqueId)
}
